plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/procedure/VacuumProcedure.java

```diff
@@ -254,8 +254,12 @@ private void doVacuum(
                     path,
                     modificationTime,
                     modificationInstant);
-            fileSystem.deleteFile(path);
-            removedFiles++;
+            filesToDeleteBuilder.add(path);
```

Comment:
Wouldn't it be safer to make sure that this list won't grow too much here? 
It comes from the iterator so we don't really know the size of it upfront and potentially it could return tens of thousands of files and I don't think we want to store such a huge list in a memory.
So maybe we can collect them up until some point and then clear the least before adding new files to it?
Something like I did here https://github.com/trinodb/trino/blob/b998f1917e16b55620522ab10b1574ae6cf1e267/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java#L1220 
WDYT ?

Reply:
Thanks for your suggestion. Actually, I was wondering if I should introduce a batch size. Added a similar logic with `1000`.